### ISSUE TITLE
Custom Blocks bug

### ISSUE THREAD

**Issue by ChefJisoo** [PluginBugs/Issues-ItemsAdder#3158]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I am sure this is a bug and it is not caused by a misconfiguration or by another plugin.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same issue was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if a solution is already known.
- [X] I already searched on the [forums](https://forum.devs.beer/) to check if anyone already has a solution for this.

### What happened?

Our custom blocks are randomly turning into chorus fruit + brown mushrooms without anyone touching them.

### Steps to reproduce the issue

Make custom blocks.
Watch them change.

### Server version

Paper 1.20.1

### ItemsAdder Version

Version 3.6.2-beta-r3

### ProtocolLib Version

The command given doesn't work, but its the latest from the website as of 16/11/23

### LoneLibs Version

The command given doesn't work, but its the latest from the website as of 16/11/23

### FULL server log

[latest (5).log](https://github.com/PluginBugs/Issues-ItemsAdder/files/13383884/latest.5.log)

**Comment by ChefJisoo** [comment]
> I cannot reproduce the issue nor find the cause. Custom blocks are not losing textures just by watching them. Also the plugin is being used by more than 5400 servers, so i'm sure that the issue is related only to some very few servers, we just need to find out the cause which unites these different setups.
> 
> This can happen:
> 
> * the server was killed and it couldn't save the chunks in time
> * itemsadder folder was deleted from the world folder
> 
> Might be to be a duplicate of this #2403 and no one find out a proper way to reproduce the issue. The only identical condition is that the servers are using CustomCrops plugin (not made by me), which might be the cause of the issue. Please let me know.

hello, thankyou for such a fast response! neither of the two things mentioned happened so I don't think it could be that. I also do not use the crops plugin mentioned so it's strange 🤔 I could later in the day attach the config I've set up for the blocks just incase I've done something wrong there?

**Comment by LoneDev6** [comment]
The other thing these servers have in common is the fact that they have a plugin which uses ItemsAdder API to manipulate blocks in the world.
Do you have any? Maybe a custom made one?

Can you attach a video of when the issue happens? As you said
> Make custom blocks.
Watch them change.

seems to be something happening right after you place a block.

**Comment by ChefJisoo** [comment]
> The other thing these servers have in common is the fact that they have a plugin which uses ItemsAdder API to manipulate blocks in the world. Do you have any? Maybe a custom made one?
> 
> Can you attach a video of when the issue happens? As you said
> 
> > Make custom blocks.
> > Watch them change.
> 
> seems to be something happening right after you place a block.

Hi! Yes it does sometimes happen right after placing one, but it also happens over the course of several hours itll just change, hence its impossible to get a video of that, but i can get a video of them changing right after placing. we also do not use any plugins that manipulate other blocks to my knowledge, certainly not the IA API

**Comment by ChefJisoo** [comment]
> The other thing these servers have in common is the fact that they have a plugin which uses ItemsAdder API to manipulate blocks in the world. Do you have any? Maybe a custom made one?
> 
> Can you attach a video of when the issue happens? As you said
> 
> > Make custom blocks.
> > Watch them change.
> 
> seems to be something happening right after you place a block.

could i adjust anything here (from config):

<html>
<body>

blocks:
--
215 | log-save-load:
216 | enabled: false
217 | min-ms: 0
218 | log-mined-unknown-blocks: true
219 | log-convert-vanilla-blocks: false
220 | log-fix-bugged-blocks: false
221 | log-unusual-block-break: false

**Comment by LoneDev6** [comment]
Yes you can enable them, but it's hard to identify this issue since I don't have a clear step-by-step way to reproduce it also on my test server.
We have to find out what is wrong in your server which is causing that.

How many players are playing while this happens?
Does it happens very frequently or rarely?

**Comment by potapenchikov** [comment]
I have noticed that this happens when the user block comes into contact with water, lava and sand physics

**Comment by LoneDev6** [comment]
This is quite strange since the plugin already handles these situations.
```yml
  fix-glitched-blocks:
    enabled: true
    only-new-chunks: false
```
This basically scans the chunk on load and fixes the blocks textures if they are wrong.

**Comment by LoneDev6** [comment]
`search-also-in-these-biomes` can be completely removed as it's not available since `3.2.3i`.
ItemsAdder will search it in any biome.

**Comment by LoneDev6** [comment]
We need a clear way to trigger this issue, without any precise way to trigger the issue I cannot fix it.
It also could be related to another plugin causing block updates when not intended.

Note: if your players don't use vanilla chorus plants you can disable their physics in Paper configuration, this will allow custom blocks to avoid losing texture in any way.
https://itemsadder.devs.beer/faq/blocks-minerals-issues/custom-blocks-glitch-texture/chorus-plants-real_transparent
